src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMBase.scala

```diff
@@ -98,14 +95,19 @@ trait LightGBMBase[TrainedModel <: Model[TrainedModel]] extends Estimator[Traine
     }
   }
 
-  protected def getTrainingCols(): Array[(String, DataType)] = {
-    val colsToCheck = Array((Some(getLabelCol), DoubleType), (Some(getFeaturesCol), VectorType),
-      (get(weightCol), DoubleType),
-      (getOptGroupCol, IntegerType), (get(validationIndicatorCol), BooleanType),
-      (get(initScoreCol), DoubleType))
-    colsToCheck.flatMap { case (col: Option[String], colType: DataType) => {
+  protected def getTrainingCols(): Array[(String, Option[DataType])] = {
+    val colsToCheck: Array[(Option[String], Option[DataType])] = Array(
+      (Some(getLabelCol), Some(DoubleType)),
+      (Some(getFeaturesCol), Some(VectorType)),
+      (get(weightCol), Some(DoubleType)),
+      (getOptGroupCol, None), // GroupCol can be any of Int/Long/String
```

Comment:
maybe instead of None, we can specify an array of those accepted types and check against them?  then you wouldn't need a filter above too

Reply:
Sure I can make that change